bundle/Command/UpdateContentVisibilityCommand.php

```diff
@@ -0,0 +1,252 @@
+        private readonly ScheduledVisibilityConfiguration $configurationService,
+        private readonly Connection $connection,
+        private readonly LoggerInterface $logger = new NullLogger(),
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this->setDescription(
+            'Updates content visibility based on publish_from and publish_to attributes and configuration.',
+        );
+        $this->addOption(
+            'limit',
+            'l',
+            InputOption::VALUE_OPTIONAL,
+            'Number of content object to process in a single iteration',
+            50,
+        );
+        $this->addOption(
+            'ttl',
+            't',
+            InputOption::VALUE_OPTIONAL,
+            'Expiration time for language caching',
+            3600,
+        );
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        $this->style = new SymfonyStyle($input, $output);
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->style->info(
+            'This command fetches content and updates visibility based on its schedule from publish_from and publish_to fields.',
+        );
+
+        if (!$this->configurationService->isEnabled()) {
+            $this->style->warning('Scheduled visibility mechanism is disabled.');
+
+            return Command::FAILURE;
+        }
+
+        $allContentTypes = $this->configurationService->isAllContentTypes();
+        $allowedContentTypes = $this->configurationService->getAllowedContentTypes();
+
+        if (!$allContentTypes && count($allowedContentTypes) === 0) {
+            $this->style->warning('No content types configured for scheduled visibility mechanism.');
+
+            return Command::FAILURE;
+        }
+
+        $query = $this->getQueryBuilder();
+
+        $contentTypeIds = [];
+        if (!$allContentTypes && count($allowedContentTypes) > 0) {
+            foreach ($allowedContentTypes as $allowedContentType) {
+                try {
+                    $contentTypeIds[] = $this->repository->getContentTypeService()->loadContentTypeByIdentifier($allowedContentType)->id;
+                } catch (NotFoundException $exception) {
+                    $this->logger->error(
+                        sprintf(
+                            "Content type with identifier '%s' does not exist: %s",
+                            $allowedContentType,
+                            $exception->getMessage(),
+                        ),
+                    );
+
+                    continue;
+                }
+            }
+        }
+
+        $this->applyContentTypeLimit($query, $contentTypeIds);
+
+        $pager = $this->getPager($query, $contentTypeIds);
+
+        if ($pager->getNbResults() === 0) {
+            $output->writeln('No content found.');
```

Comment:
We should also use `$this->style` here.

Reply:
fixed in b325c484488f68c043f8249a60b0d515697f35f5